CONTRIBUTING.md

```diff
@@ -25,22 +25,535 @@ Below is a sample of how additional vehicle directories should be created.
+| boards | PCB schematics and board designs for control modules that are integrated with the vehicle, typically these are the throttle, steering and brake boards, but can include other items should they be contributed. |
+| control | User Space applications to provide mechanisms for the user to control the vehicle directly via something like a PC.  Applications can be command-line or GUI based and include items like a joystick controller |
+| firmware | Arduino code for the throttle, steering and brake functionality.  This directory also includes code for the CAN gateway module, which uses a stock Arduino board. The firmware should remain modular and portable to a a different Arduino board |
+| vehicle_info | Information on specific vehicles such as sensor outputs and wiring diagrams.  Each additional vehicle should be in its own subdirectory |
+
+## Diagrams
+
+Adding diagrams for new boards or other components should be located in the directory most appropriate for their function:
+* A new enclosure should go in the 3d_models directory
+* A new board diagram should go in the boards directory
+* Any new vehicle information should reside in the vehicle_info directory
+
+### Editing diagrams
+
+Existing diagrams should be edited using the Autodesk Eagle as mentioned above.
+
+## Building
+
+To build the various components in the system, please see the README.md
+
+## Documentation
+
+* Documentation updates are welcome
+* Documentation should be located in the *doc* directory where it is most applicable (See the Directory descriptions above)
+	* If the *doc* directory does not exist, please create it as part of the submission
+* Documentation pull requests function in the same way as other pull requests
+
+### Documentation Style
+
+* Documentation should be in markdown format
+
+## Helpful Links
+
+* [Autodesk Eagle](https://cadsoft.io)
+* [Arduino FAQ](https://www.arduino.cc/en/Main/FAQ)
+* [Throttle/Steering/CAN Gateway board](https://www.arduino.cc/en/Main/ArduinoBoardUno)
+* [Braking board](https://www.arduino.cc/en/Main/arduinoBoardMega2560)
+
+# OSCC Coding standard
+
+As this is an automotive initiative, this coding standard is based on
+the MISRA C-2012 standard. The reasoning for the given rule or directive in the
+coding standard is not included in this as it would create much too large a
+document.
+
+The MISRA standard is more exacting that the OSCC standard. It is not
+implied that compliance with this standard guarantees compliance with
+the MISRA standard.
+
+## 1. Directives
+
+1. Implemented behavior must be documented and understood
+1. Comply with the C99 definition of the standard
+1. All source files must compile without errors
+1. Run-time failures must be limited
+	1. All arithmetic operations must be checked (e.g. divide by zero)
+	1. Pointer arithmetic must be checked
+	1. Array bounds must be checked
+	1. Check validity of parameters either before or inside a function call
+	1. Check pointer validity before dereference

```

Comment:
For consistent voice, number 4 & 5 could be reworded to 'Validity of parameters must be checked either before... and Pointer validity must be checked before dereference'